### ISSUE TITLE
setting individual learning rate for specific tensors

### ISSUE THREAD

**Issue by KayneWest** [tinygrad/tinygrad#1145]
Would adding this attribute to individual `Tensors` be useful to anyone else? I've added it to my local `tinygrad` for a model I'm hacking on, but more than happy to submit a PR if this doesn't break anything in ways unforeseen to myself

add `lr_mult` to Tensor
```
class Tensor:
  __slots__ = "lazydata", "requires_grad", "grad", "_ctx", "lr_mult"
  __deletable__ = ('_ctx',)
  training: ClassVar[bool] = False
  no_grad: ClassVar[bool] = False
  default_type: ClassVar[DType] = dtypes.float32

  def __init__(self, data:Union[int, float, list, tuple, LazyBuffer, np.ndarray], device:Optional[str]=None, dtype:Optional[DType]=None, requires_grad:Optional[bool]=None):
    assert dtype is None or isinstance(dtype, DType), f"invalid dtype {dtype}"
    device = Device.canonicalize(device)
    # tensors have gradients, buffers do not
    self.lr_mult = float(1.0)
   ...
```
use `lr_mult` in the optimizer
```
class LAMB(Optimizer):
  def __init__(self, params: List[Tensor], lr=0.001, b1=0.9, b2=0.999, eps=1e-6, wd=0.0, adam=False):
    super().__init__(params)
    self.lr, self.b1, self.b2, self.eps, self.wd, self.adam, self.t = lr, b1, b2, eps, wd, adam, Tensor([0], requires_grad=False).realize()
    self.m = [Tensor.zeros(*t.shape, device=t.device, requires_grad=False) for t in self.params]
    self.v = [Tensor.zeros(*t.shape, device=t.device, requires_grad=False) for t in self.params]

  def step(self) -> None:
    self.t.assign(self.t + 1).realize()
    for i, t in enumerate(self.params):
      assert t.grad is not None
      g = t.grad.realize()
      self.m[i].assign(self.b1 * self.m[i] + (1.0 - self.b1) * g).realize()
      self.v[i].assign(self.b2 * self.v[i] + (1.0 - self.b2) * (g * g)).realize()
      m_hat = self.m[i] / (1.0 - self.b1**self.t)
      v_hat = self.v[i] / (1.0 - self.b2**self.t)
      up = (m_hat / (v_hat.sqrt() + self.eps)) + self.wd * t.detach()
      if not self.adam:
        r1 = t.detach().square().sum().sqrt()
        r2 = up.square().sum().sqrt()
        r = Tensor.where(r1 > 0, Tensor.where(r2 > 0, r1 / r2, 1.0), 1.0)
      else:
        r = 1.0
      t.assign(t.detach() - (self.lr * t.lr_mult) * r * up)
    self.realize([self.t] + self.m + self.v)

```
use `lr_mult` to dampen effects on pretrained layers
```
# Use a smaller learning rate for pre-trained convolutional layers.

from tinygrad.state import get_parameters
from models.resnet import ResNet50
from custom_model import EmbeddingNet

feature_net = ResNet50()
feature_net.load_from_pretrained()

feature_params = get_parameters(feature_net)
for param in feature_params:
    setattr(param, 'lr_mult', 0.01)

embedding_net = EmbeddingNet(feature_net)
```

**Comment by KayneWest** [comment]
Closing. Added PR here. https://github.com/tinygrad/tinygrad/pull/1147
